### PR TITLE
Combining syntax cheat sheet and full syntax reference pages

### PR DESCRIPTION
**Pull request by jingting1412**
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Implements the changes as discussed in issue #2091, more specifically I'm implementing the points as summarised [here](https://github.com/MarkBind/markbind/issues/2091#issuecomment-1909719113) by @kaixin-hc 

**Anything you'd like to highlight/discuss:**
Open to any discussion regarding how best to structure/implement the syntax reference page. 

**Testing instructions:**
Load the UG and ensure changes are according to what is discussed.

**Proposed commit message: (wrap lines at 72 characters)**
Combine syntax cheat sheet and full syntax reference pages

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

File: docs/userGuide/syntaxReference.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<frontmatter>
+  title: "User Guide: Syntax Reference"
+  layout: userGuide.md
+  pageNav: 5
+</frontmatter>
+
+# Syntax Reference
+
+<box type="info">
+  Expand the panels for detailed explanations of the corresponding feature! Click on the link on the header to go to the corresponding page!
```
Comment (kaixin-hc) [inline docs/userGuide/syntaxReference.md:10]:
```suggestion
Expand the panels for detailed explanations of the feature or click on the link in the header to go to the corresponding page!
```

**Review by kaixin-hc — COMMENTED** [review]
page is looking great! here are a couple of things to check on

File: docs/userGuide/readerFacingFeatures.md [hunk 2]
```diff
@@ -7,9 +7,8 @@
 
 # Reader-Facing Features
 
-<include src="fullSyntaxReference.md#dummy" optional />
 
-{% from "userGuide/fullSyntaxReference.md" import syntax_topics as topics %}
```
[the remark this answers is not in the document]
Comment (jingting1412) [reply docs/userGuide/readerFacingFeatures.md:12]: Yes sorry for my mistake, I've since updated the code!

[the remark this answers is not in the document]
Comment (jingting1412) [reply docs/userGuide/readerFacingFeatures.md:12]: On an unrelated note, I think this page can benefit from a site nav, maybe I can open an issue to quickly add it in?

[the remark this answers is not in the document]
Comment (kaixin-hc) [reply docs/userGuide/readerFacingFeatures.md:12]: Yes, new issue and minor change sounds good. While you're at it, I think a tip box at the top (similar to the syntax reference page) wouldn't hurt

**Review by kaixin-hc — APPROVED** [review]
LGTM! thanks for reworking this, I think it will be nice from a user perspective ❤️

**Pull request merged**